scalding-core/src/main/scala/com/twitter/scalding/filecache/DistributedCacheFile.scala

```diff
@@ -105,16 +105,14 @@ final case class UncachedFile private[scalding] (source: Either[String, URI]) {
   }
 
   private[this] def addHdfs(conf: Configuration): CachedFile = {
-    HDistributedCache.createSymlink(conf)
```

Comment:
HDistributedCache.createSymlink is from hadoop-mapreduce-client-core-2.5.0.jar

We have specified the below version dependency on hadoop in build.sbt for our version of Scalding: 
> val hadoopVersion = "2.5.0"

Do we need to maintain backward compatibility to older versions of hadoop as well?